UnitTests/PaymentsCoreTests/APIClient_Tests.swift

```diff
@@ -2,33 +2,38 @@ import XCTest
 @testable import PaymentsCore
 @testable import TestShared
 
+// swiftlint:disable force_unwrapping implicitly_unwrapped_optional
 class APIClient_Tests: XCTestCase {
 
     // MARK: - Helper Properties
-    let mockClientID = "mockClientId"
-    let mockAccessToken = "mockAccessToken"
 
-    // swiftlint:disable:next force_unwrapping
     let successURLResponse = HTTPURLResponse(url: URL(string: "www.test.com")!, statusCode: 200, httpVersion: "https", headerFields: [:])
     let fakeRequest = FakeRequest()
 
-    // swiftlint:disable implicitly_unwrapped_optional
-    var config: CoreConfig!
+    let config = CoreConfig(accessToken: "mockAccessToken", environment: .sandbox)
     var mockURLSession: MockURLSession!
-    var apiClient: APIClient!
-    // swiftlint:enable implicitly_unwrapped_optional
+    var sut: APIClient!
+    var mockHTTP: MockHTTP!
 
     // MARK: - Test lifecycle
 
     override func setUp() {
         super.setUp()
-        config = CoreConfig(accessToken: mockAccessToken, environment: .sandbox)
         mockURLSession = MockURLSession()
-        mockURLSession.cannedError = nil
-        mockURLSession.cannedURLResponse = nil
-        mockURLSession.cannedJSONData = nil
-
-        apiClient = APIClient(urlSession: mockURLSession, coreConfig: config)
+        mockHTTP = MockHTTP(urlSession: mockURLSession, coreConfig: config)
```

Comment:
do we still need `MockUrlSession` in the test? `APIClient` now interacts directly with `HTTP`, so the  `URLSession` is an internal component of `HTTP` that `APIClient` doesnt know about.
`MockHTTP` could have its own init with no params that calls super with the corresponding mocks.

Reply:
Yes, that is desired end state! However there is a limitation in our current implementation that prevents us from fully mocking `HTTP.fetch()` without needing to call a super ([here](https://github.com/paypal/iOS-SDK/blob/e376d38bb270bd95f6e80b5e44cb6e7225f3e5b7/UnitTests/TestShared/MockHTTP.swift#L10-L14)). 

This is due to the use of the generic return type `T.ResponseType`. We can't instantiate it, therefore can't make a fake return value in the mock. So - currently, MockHTTP isn't really a _true mock_ since it calls the super and calls the actual method, which is why we end up still needing to stub the URLSession response. 

I was toying with addressing this now, but I think it would lead to a larger refactor of the `APIRequest` protocol that we use heavily. We can make a ticket to 🏈 punt on this.

Let me know if that makes sense!

Reply:
Ticket created! DTNOR-811